lib/matplotlib/tests/test_category.py

```diff
@@ -187,6 +187,65 @@ def test_plot_1d_missing(self):
         self.axis_test(ax.yaxis, self.dmticks, self.dmlabels, self.dmunit_data)
 
     @cleanup
+    @pytest.mark.usefixtures("data")
+    def test_plot_bytes(self):
+        counts = np.array([4, 6, 5, 1])
+        fig, ax = plt.subplots(ncols=3)
+
+        ax[0].bar(self.d, counts)
+
+        types = [v.encode('ascii') for v in self.d]
+        ax[1].bar(types, counts)
+
+        types = np.array(types)
+        ax[2].bar(types, counts)
+
+        fig.canvas.draw()
+
+        # All three plots should look like the string one.
+        self.axis_test(ax[1].xaxis,
+                       ax[0].xaxis.get_majorticklocs(),
+                       lt(ax[0].xaxis.get_majorticklabels()),
+                       ax[0].xaxis.unit_data)
+        self.axis_test(ax[2].xaxis,
+                       ax[0].xaxis.get_majorticklocs(),
+                       lt(ax[0].xaxis.get_majorticklabels()),
+                       ax[0].xaxis.unit_data)
+
+    @cleanup
+    def test_plot_numlike(self):
+        counts = np.array([4, 6, 5, 1])
+        fig, ax = plt.subplots(ncols=4)
+
+        types = ['1', '11', '3', '1']
+        ax[0].bar(types, counts)
+
+        types = np.array(types)
+        ax[1].bar(types, counts)
+
+        types = [b'1', b'11', b'3', b'1']
+        ax[2].bar(types, counts)
+
+        types = np.array(types)
+        ax[3].bar(types, counts)
+
+        fig.canvas.draw()
+
+        # All four plots should look like the string one.
+        self.axis_test(ax[1].xaxis,
```

Comment:
same as above, the expected ticklocs and ticklabels are [0, 1, 2, 3] and ['1', '11', '3', '1'] respectively, The axis unit data should be mocked, as seen in line 141.